### PR TITLE
Added Django 4.0 support.

### LINKED ISSUE #86: Django 4.0 ImportError: cannot import name 'force_text' from 'django.utils.encoding'
I upgraded Django to 4.0 and I'm getting an error:
```
  File "//easy_select2/__init__.py", line 1, in <module>
    from easy_select2.utils import select2_modelform_meta, select2_modelform, \
  File "easy_select2/utils.py", line 5, in <module>
    from easy_select2 import forms as es2_forms
  File "easy_select2/forms.py", line 4, in <module>
    from django.utils.encoding import force_text
ImportError: cannot import name 'force_text' from 'django.utils.encoding' (django/utils/encoding.py)
```

We need to change force_text to force_str

### PR DESCRIPTION
**Pull request by mikekeda**
Fix #86 and #87 
Also I added python 3.9 / 3.10 (I could remove this if needed)

### CODE REVIEW

**Comment by gauravv7** [comment]
Thanks for the support! Let me review this.

**Comment by asyncee** [comment]
Lgtm.

@gauravv7 what you think?

@mikekeda thanks!

File: tox.ini [hunk 1]
```diff
@@ -21,6 +22,8 @@ deps =
     django21:  django>=2.1,<2.2
     django22:  django>=2.2,<2.3
     django30:  django>=3.0,<3.1
+    django31:  django>=3.1,<4.0
+    django40:  django>=4.0,<5.0
```
Comment (hartwork) [inline tox.ini:26]: Is it intentional that Django 3.2 is missing here?

File: setup.py [hunk 2]
```diff
@@ -64,6 +64,8 @@ def run_tests(self):
           'Programming Language :: Python :: 3.6',
           'Programming Language :: Python :: 3.7',
           'Programming Language :: Python :: 3.8',
+          'Programming Language :: Python :: 3.9',
+          'Programming Language :: Python :: 3.10',
```
Comment (hartwork) [inline setup.py:68]: The CI is only covering up to Python 3.9 above. Is that asymmetry intended?

To make Python 3.10 work in the CI, and update from bionic/18.04 to hirsute/21.04 may be needed. 

PS: Python 3.5 and 3.6 support can be dropped by now, as far as I am concerned.

File: tox.ini (hunk 1 above)
Comment (mikekeda) [reply tox.ini:26]: I assumed that 3.2 will be covered by 3.1,<4.0 range,
Should I split this range into 2 ranges (>=3.1,<3.2 and >=3.2,<4.0)?

File: setup.py (hunk 2 above)
Comment (mikekeda) [reply setup.py:68]: I tested those changes on 3.10
Should I add python 3.10 to .travis.yml?

File: tox.ini (hunk 1 above)
Comment (hartwork) [reply tox.ini:26]: I see. `django>=3.1,<4.0` will give you 3.2 then and 3.1 will not be covered. If you want both to be covered, splitting sounds like a good idea.

File: setup.py (hunk 2 above)
Comment (hartwork) [reply setup.py:68]: My _personal_ vote would be "yes please, that would be welcome". @gauravv7 are there any reasons to stick to bionic for longer?

Comment (hartwork) [reply setup.py:68]: PS: I noticed use of `ppa:deadsnakes/ppa` just now. I imagine that repository _may_ even allow use of python3.10 even with bionic, just to add to the pool of options.

File: tox.ini (hunk 1 above)
Comment (hartwork) [reply tox.ini:26]: PS: Based on the image below, maybe Django 2.2 + 3.2 + 4.0 is all versions worth covering in 2021-12. 

Django releases

**Comment by gauravv7** [comment]
@asyncee yes this looks to me too, I also finished my review.

@hartwork that version had some issue with tox tests and pypi pull of django. (Exact error I need to check)

File: setup.py (hunk 2 above)
Comment (gauravv7) [reply setup.py:68]: Yes, the tox tests were failing at that time, Travis build was failing due to timeout while setting up tox to run tests. So I ran based on bionic.

Comment (gauravv7) [reply setup.py:68]: @mikekeda @hartwork we can add 3.10 in tox and Travis configs

File: tox.ini (hunk 1 above)
Comment (mikekeda) [reply tox.ini:26]: so no need to split the range, right?

**Comment by gauravv7** [comment]
something is failing while running checks

https://github.com/asyncee/django-easy-select2/runs/4586197919?check_suite_focus=true

PS: @mikekeda

**Comment by mikekeda** [comment]
The build failed with:
```
Run actions/setup-python@v1.1.1
Error: Unable to process command '##[set-env name=pythonLocation;]/opt/hostedtoolcache/Python/3.8.12/x64' successfully.
Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
```
I will try to update actions/setup-python to v1.2.3

File: tox.ini (hunk 1 above)
Comment (hartwork) [reply tox.ini:27]: It should probably be `django32:  django>=3.2,<3.3` though, then. What do you think?

Comment (mikekeda) [reply tox.ini]: Replaced django31 with django32

**Comment by hartwork** [comment]
> The build failed with:
> 
> ```
> Run actions/setup-python@v1.1.1
> Error: Unable to process command '##[set-env name=pythonLocation;]/opt/hostedtoolcache/Python/3.8.12/x64' successfully.
> Error: The `set-env` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
> ```
> 
> I will try to update actions/setup-python to v1.2.3

@mikekeda allow me to braindump about that:
- You will likely need a fix like https://github.com/hartwork/jawanndenn/commit/f65988b8a4ed18c0a2cae4c5d2159eeb3704069d
- My vote for fixing that in a dedicated pull request. Can be merged in no time. Just an idea!
- We should make GitHub Dependabot keep GitHub Actions up-to-date. I can volunteer for a pull-request, unless someone else wants to play with that.
- We should probably add cron mode for the CI. It's a pity that this issue has not shown earlier and is distracting your from your actual pull request. Sorry about that. Happy to do a PR unless someone else would rather do it themselves.

Let me know what you think and how to proceed :beers:

File: tox.ini [hunk 3]
```diff
@@ -21,6 +23,8 @@ deps =
     django21:  django>=2.1,<2.2
     django22:  django>=2.2,<2.3
     django30:  django>=3.0,<3.1
+    django32:  django>=3.2,<4.0
+    django40:  django>=4.0,<5.0
```
Comment (hartwork) [inline tox.ini:27]: The upper bound doesn't seem right. Expecting `<3.3` and `<4.1` here. Otherwise these targets are "moving too much" over time. Thanks!

Comment (mikekeda) [reply tox.ini]: Fixed

Comment (hartwork) [reply tox.ini]: Thank you!

**Comment by mikekeda** [comment]
https://github.com/asyncee/django-easy-select2/actions/runs/1603565588
@hartwork this build succeeded, I'm assuming that upgrading actions/setup-python fixed this

**Comment by hartwork** [comment]
@mikekeda right, that fix I mentioned is needed on their side not ours, we already have it fixed. So good point and idea. Version `v2.3.1` seems like the best target to me.

**Comment by hartwork** [comment]
@mikekeda could you rebase onto latest `master`? With PR #89 and PR #90 out of the way, you should be able to focus on Django 4 and Python version support now. Looking forward to that.

**Comment by mikekeda** [comment]
@hartwork I merged master

**Comment by hartwork** [comment]
Does anyone have a link to where current Travis runs can be viewed? I looked and couldn't find it.
(I'm thinking we _may_ need to migrate the Travis bits to GitHub Actions, as well. In a dedicated pull request.)

**Comment by hartwork** [comment]
Found it at https://travis-ci.org/github/asyncee/django-easy-select2/builds now. Reads "read-only mode" and last run things 6 months ago… If anything has been migrated from travis-ci.org to travis-ci.com, it doesn't seem to be visible to the public.

**Comment by gauravv7** [comment]
@hartwork actually due to covid, I was not able to put up anything, so travis would actually not have anything.

**Comment by gauravv7** [comment]
@mikekeda thanks for this beautiful PR. Merging it.

**Comment by hartwork** [comment]
>> Found it at https://travis-ci.org/github/asyncee/django-easy-select2/builds now. Reads "read-only mode" and last ran things 6 months ago… If anything has been migrated from travis-ci.org to travis-ci.com, it doesn't seem to be visible to the public.
>
> @hartwork actually due to covid, I was not able to put up anything, so travis would actually not have anything.

I will try a PR on that migration then. With some luck: later today. I will omit coveralls for someone to add after me. I hope that's okay.

**Comment by asyncee** [comment]
I agree on discarding Travis. Please, ping me when i should remove project from travis-ci.org. Thanks!

**Comment by gauravv7** [comment]
@hartwork  no worries, I will add coveralls but yes your PR would be always helpful :)

**Comment by gauravv7** [comment]
@asyncee waiting for @hartwork PR and then let's delete after merging the PR.

**Pull request merged**